libs/domain/picking/offline/services/adapter/picking-list-offline.adapter.ts

```diff
@@ -50,18 +50,29 @@ export class PickingListOfflineAdapter implements PickingListAdapter {
               const andCollections: Collection<PickingListSerialized>[] = [];
 
               if (qualifier.status) {
-                andCollections.push(
-                  pickingListStore.where('localStatus').equals(qualifier.status)
-                );
-              }
-
-              if (qualifier.orderReferences) {
-                andCollections.push(
-                  pickingListStore
-                    .where('orderReferences')
-                    .startsWithAnyOf(qualifier.orderReferences)
-                    .distinct()
-                );
+                if (
+                  qualifier.searchOrderReference &&
+                  qualifier.searchOrderReference.length >= 2
+                ) {
+                  const searchValue = qualifier.searchOrderReference;
+                  andCollections.push(
+                    pickingListStore
+                      .where('localStatus')
+                      .equals(qualifier.status)
+                      .and((record: PickingListSerialized) =>
+                        record.orderReferences.some((ref) =>
+                          ref.startsWith(searchValue)
+                        )
```

Comment:
This is not good for performance as `and()` operation is executed on the whole collection on JS side returned by the `where()` operation which is much wider.
You should not mix the 2 queries and have the search by order refs done as part of `where()` operation so it's carried by the indexed-db engine directly and not JS. Just push them to `andCollections` separately which then will be filtered by intersection on JS side.

Reply:
Updated after our 1on1 alignment